mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/cfg/UuidDataType.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.orm.coordination.outboxpolling.cfg;
+
+import static org.hibernate.search.mapper.orm.coordination.outboxpolling.cfg.HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_OUTBOX_EVENT_UUID_DATA_TYPE;
+
+import java.lang.invoke.MethodHandles;
+
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.SQLServerDialect;
+import org.hibernate.search.mapper.orm.coordination.outboxpolling.logging.impl.Log;
+import org.hibernate.search.util.common.annotation.Incubating;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+@Incubating
+public enum UuidDataType {
+	/**
+	 * Is database specific. Some databases might not work correctly with a particular representation of UUID hence the
+	 * default provides the value against which particular database was tested. If the database is not on the testing list
+	 * the default would use a character representation as, characters should be overall supported by a database at least
+	 * as a type.
+	 * <p>
+	 * Note that in case of MS SQL database using {@link #CHAR} representation might lead to undesired behavior and
+	 * occurrence of deadlocks.
+	 */
```

Comment:
I suggested a few minor improvements (it needs to be reformatted as I added/removed some content):
```suggestion
	/**
	 * Database specific. Some databases might not work correctly with a particular representation of UUID hence the
	 * default provides a sensible value depending on which particular database is used. By default, a character representation is used as characters should be overall supported by databases.
	 * <p>
	 * But for some databases, a different value is used. For instance, in the case of Microsoft SQL Server, using a character representation might lead to undesired behavior in the form of deadlocks so a binary representation is used by default.
	 */
```